KeyedSemaphores/KeyedSemaphoresCollection.cs

```diff
@@ -335,5 +335,22 @@ public bool TryLock(TKey key, TimeSpan timeout, Action callback, CancellationTok
 
             return true;
         }
+
+        /// <summary>
+        ///     Check if keyed semaphore already has the provided unique key
+        /// </summary>
+        /// <param name="key">
+        ///     The unique key of this keyed semaphore
+        /// </param>
+        /// <returns>
+        ///     True when key are already locked
+        ///     False when key are available for lock
+        /// </returns>
+        public bool ContainsKey(TKey key)
+        {
+            if (key == null) throw new ArgumentNullException(nameof(key));
+
+            return Index.ContainsKey(key);
```

Comment:
Is this the desired behavior? Because you are assuming that the index will always be cleaned up immediately after the last usage - which is true today - but it does not necessarily mean that someone is locking this keyed semaphore.

Reply:
What do you mean? How it could be still locked? If it locked means i can't lock it again, isn't true?

as I expected
I need to get to know if key already locked and let know others about that like throw exception or skip that part of job

as example where **_semaphore.CurrentCount == 0** same as **Contains(key)**
```
class ImportantJob
{
    private static readonly SemaphoreSlim _semaphore = new SemaphoreSlim(1, 1);

    public async Task ImportantUpdate()
    {
        if (_semaphore.CurrentCount == 0)
        {
            _logger.LogInformation("Some important already busy, come next time");
            return;
        }

        await _semaphore.WaitAsync();
        try
        {
            _logger.LogInformation("Updating something important...");
        }
        finally
        {
            _semaphore.Release();
        }
    }
}
```

Reply:
The ideal thing to do is to give it a more generic name instead of `ContainsKey`. I have made a similar libary and have an identical method that's called `IsInUse`. That way, if the implementation changes one day (for example an option to retain items in the dictionary) it will still be suitably named.

Reply:
Make sense, renamed